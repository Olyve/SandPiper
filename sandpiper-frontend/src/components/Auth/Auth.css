.auth {
  display: flex;
  flex-direction: column;
  height: 100%;
  align-items: center;
  justify-content: center;
}

.message {
  display: flex;
  flex-direction: row;
  height: 60px;
}

.message a {
  color: blue;
  text-decoration: none;
  margin-top: 16px;
  margin-bottom: 16px;
  margin-left: 5px;
}

.message a:hover {
  text-decoration: underline;
  cursor: default;
}