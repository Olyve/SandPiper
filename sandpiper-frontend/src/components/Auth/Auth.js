import React, {Component} from 'react';
import Signup from './Signup';
import Login from './Login';
import './Auth.css';


class Auth extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showLogin: true
    };

    this.toggleView = this.toggleView.bind(this);
  }

  toggleView() {
    this.setState({showLogin: !this.state.showLogin});
  }

  createMessage(message) {
    return (
      <div className='message'>
        <p>
          {message}
          <a onClick={this.toggleView}>click here</a>.
        </p>
      </div>
    );
  }

  render() {
    var component;
    var message;

    if (this.state.showLogin) {
      component = <Login handleLogin={this.props.handleLogin} />;
      message = this.createMessage('If you don\'t have an account');
    }
    else {
      component = <Signup handleSignup={this.props.handleSignup} />;
      message = this.createMessage('If you already have an account');
    }

    return (
      <div className='auth'>
        {component}
        {message}
      </div>
    );
  }
}


export default Auth;