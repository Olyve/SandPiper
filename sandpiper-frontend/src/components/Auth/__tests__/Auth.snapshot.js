import React from 'react';
import Auth from '../Auth';

describe('Auth', () => {
  it('should render Login by default', () => {
    const wrapper = shallow(
      <Auth />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render Register when showLogin is false', () => {
    const wrapper = shallow(
      <Auth />
    );

    // Find and trigger the button to toggle state
    let submit = wrapper.find('a');
    submit.simulate('click', { preventDefault: () => {} });

    expect(wrapper).toMatchSnapshot();
  });
});