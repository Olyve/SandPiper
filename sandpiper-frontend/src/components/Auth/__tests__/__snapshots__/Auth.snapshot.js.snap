// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should render Login by default 1`] = `
<div
  className="auth"
>
  <Login />
  <div
    className="message"
  >
    <p>
      If you don't have an account
      <a
        onClick={[Function]}
      >
        click here
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`Auth should render Register when showLogin is false 1`] = `
<div
  className="auth"
>
  <Signup />
  <div
    className="message"
  >
    <p>
      If you already have an account
      <a
        onClick={[Function]}
      >
        click here
      </a>
      .
    </p>
  </div>
</div>
`;
